### ISSUE TITLE
Blank .bm bookmarking request message

### ISSUE THREAD

**Issue by Owez** [python-discord/sir-lancebot#338]
As an alternative from explicitly giving the message ID to bookmark when using the `.bm [message id]` command, add a shorthand version with the entire command being `.bm`. This will work by simply getting the message saying "`.bm`"'s ID and bookmarking that. For example:

```none
USER1 [m_id 334]: This is a cool post
USER2 [m_id 942]: .bm
```

In this example, instead of USER2 writing `.bm 334` to bookmark the message ID, they can simply type `.bm` and bookmark their own message (message ID of `942`). This can be useful for easier bookmarking with developer/debug mode disabled.

**Comment by kwzrd** [comment]
I've also seen a suggestion that bot could listen for further clicks on its :incoming_envelope: reaction, and dispatch the bookmark to those who click as well. This would be a super clean way to ask to be given the bm as well, although it may not be very obvious that it does that.

**Comment by LordHemlock** [comment]
> I've also seen a suggestion that the bot could listen for further clicks on its 📨 reaction, and dispatch the bookmark to those who click as well. This would be a super clean way to ask to be given the bm as well, although it may not be very obvious that it does that.

It'd be more obvious if we also added a little footer to the embed that mentions that.  That's actually a really cool idea.  And would be relatively easy to implement.  

The only thing we'll have to work out is making sure people don't accidentally spam themselves by clicking then unclicking then clicking rinse and repeat.  Although I guess in fairness they'd be doing it to themselves.

**Comment by Akarys42** [comment]
Another thing that would be quite cool is to be able to delete a bookmark, since there is no way of doing it right now. We could just have a `.bm delete [id]` command, and the bot will search through the DMs and delete it. Another thing is that if the original message is deleted, the bookmark stays, but I don't think that's a real problem.

**Comment by scOwez** [comment]
If the bot sent a message itself (as what @LordHemlock said), the shorthand `.bm` could bookmark the bot's own message, preventing any deletions.

**Comment by AG4lyf** [comment]
Can i work on it , i am nearly done.

**Comment by AG4lyf** [comment]
Also we can fetch the 334 id message
The way you are asking if you need to add hints you have to do `.bookmark * hints are here`
then why not use the message above that bookmark message
also it keep the way clean because one can react to that message and get same bookmark too.

**Comment by ikuyarihS** [comment]
So what was the conclusion of the discussion?

Right now PR #386 is bookmarking the message invoking the command, will send this to the command invoker as well as everyone who reacted with the `pin` message:

![image](https://user-images.githubusercontent.com/19921765/77995495-1383ae00-7356-11ea-9b1d-8e6af55f2ff1.png)

I do not think this is a good design, because everytime I look at it, I have no idea what it is bookmarking.

So, what should the bot bookmark when someone invokes `.bm` in this case?

**Comment by AG4lyf** [comment]
![image](https://user-images.githubusercontent.com/45149585/77995767-02cd3b00-734a-11ea-8c28-4e125ba6e0c2.png)
as the example given here we can do is bookmark the message with id 334, and one can always get the same bookmark by reacting.

**Comment by ikuyarihS** [comment]
What do you refer here as "same bookmark" though? Because in everyone's DM they will see a single `.bm` like in the image above, which, 3 days later, will make no sense to them.

**Comment by AG4lyf** [comment]
no i mean bookmark whose target message is m_id 334

**Comment by ikuyarihS** [comment]
In that case are you going to bookmark the message with content `This is a cool post` or the message with `.bm`? Because right now the PR is bookmaking the one with `.bm` instead.

**Comment by AG4lyf** [comment]
`This is a cool post` one, I can make some changes in PR. also that one will make more sense.

**Comment by ikuyarihS** [comment]
In that case, what if the chat is like this
```
USER1 [id 1]: This is a cool post
USER2 [id 2]: This is cool! I love it
USER3 [id 3]: Nice one!
USER2 [id 2]: .bm
```
Which message will be bookmarked?

**Comment by AG4lyf** [comment]
id 3 
ofc
`USER3 [id 3]: Nice one!`

**Comment by AG4lyf** [comment]
Ok after internal talking that we need to use union
so now 
.bm @USER1 
will bookmark the last message by USER1 in that channel
also that way you can use hints also

**Comment by AG4lyf** [comment]
I just got a really good idea. I saw something similar to this in one of my server . Here is link how it looks in working
https://ag-1436.xyz/indinet

**Comment by sco1** [comment]
Very funny. Please don't shitpost in GH issues.

**Comment by kwzrd** [comment]
Issue is now available if anyone wants to work on it.

Contributions should take into account reviewer feedback from the two PRs:
https://github.com/python-discord/seasonalbot/pull/349
https://github.com/python-discord/seasonalbot/pull/386

**Comment by ChrisLovering** [comment]
A version of this was implemented in #645. If further changes wish to be made to this command, we should open a new issue for clarity.
